src/inputs/UploadFile/UploadFile.js

```diff
@@ -36,18 +38,22 @@ export const UploadFile = (props) => {
   const classes = useStyles();
 
   // TODO: create a generic component to truncate texts in Typography elements
-  const smartFileName = useMemo(() => {
-    const fileNameLength = file?.name?.length ?? 0;
+  const fileNameElement = useMemo(() => {
+    let fullLabel = file?.name;
+    if (shouldHideFileName) {
+      const extension = PathUtils.getExtensionFromFileName(file?.name)?.toUpperCase();
+      fullLabel =
+        labels?.getFileNamePlaceholder !== undefined ? labels.getFileNamePlaceholder(extension) : `${extension} file`;
+    }
+
     const fileName = (
       <Typography data-cy="file-name" noWrap sx={{ maxWidth: '20ch' }}>
-        {file.name}
+        {fullLabel}
       </Typography>
     );
 
-    return (
-      <Stack>{fileNameLength > 20 ? <FadingTooltip title={file.name}>{fileName}</FadingTooltip> : fileName}</Stack>
-    );
-  }, [file.name]);
+    return fullLabel?.length ?? 0 > 20 ? <FadingTooltip title={fullLabel}>{fileName}</FadingTooltip> : fileName;
```

Comment:
Maybe set the label maximum length in a const at the top of the file ?

Reply:
Even better it should probably be a prop, but it would mean to modify the unit tests and replace it everywhere the value is used (e.g. several lines above, in `maxWidth: '20ch'`). This would take some time: too much time for something that is out of scope of the current feature (the "20" value has not been added by the current branch) :slightly_smiling_face: